chart/openfaas/templates/gateway-dep.yaml

```diff
@@ -159,6 +166,75 @@ spec:
         ports:
         - containerPort: 8081
           protocol: TCP
+      {{- else if .Values.faaslambda.create }}
+      - name: faaslambda
+        resources:
+          {{- .Values.operator.resources | toYaml | nindent 12 }}
+        image: {{ .Values.faaslambda.image }}
+        imagePullPolicy: {{ .Values.openfaasImagePullPolicy }}
+        {{- if .Values.securityContext }}
+        securityContext:
+          readOnlyRootFilesystem: true
+          runAsUser: 10001
+        {{- end }}
+        env:
+          - name: port
+            value: "8081"
+          - name: function_namespace
+            value: {{ $functionNs | quote }}
+          - name: read_timeout
+            value: "{{ .Values.faaslambda.readTimeout }}"
+          - name: write_timeout
+            value: "{{ .Values.faaslambda.writeTimeout }}"
+          - name: image_pull_policy
+            value: {{ .Values.faaslambda.imagePullPolicy | quote }}
+          - name: http_probe
+            value: "{{ .Values.faaslambda.httpProbe }}"
+          - name: secret_mount_path
+            value: "/var/secrets"
+          {{- if .Values.basic_auth }}
+          - name: basic_auth
+            value: "{{ .Values.basic_auth }}"
+          {{- end }}
+          - name: readiness_probe_initial_delay_seconds
+            value: "{{ .Values.faaslambda.readinessProbe.initialDelaySeconds }}"
+          - name: readiness_probe_timeout_seconds
+            value: "{{ .Values.faaslambda.readinessProbe.timeoutSeconds }}"
+          - name: readiness_probe_period_seconds
+            value: "{{ .Values.faaslambda.readinessProbe.periodSeconds }}"
+          - name: liveness_probe_initial_delay_seconds
+            value: "{{ .Values.faaslambda.livenessProbe.initialDelaySeconds }}"
+          - name: liveness_probe_timeout_seconds
+            value: "{{ .Values.faaslambda.livenessProbe.timeoutSeconds }}"
+          - name: liveness_probe_period_seconds
+            value: "{{ .Values.faaslambda.livenessProbe.periodSeconds }}"
+          - name: lambda_execution_role
+            value: "{{ .Values.faaslambda.lambda_execution_role }}"
+          - name: access_token
+            value: "{{ .Values.faaslambda.access_token }}"
```

Comment:
I wonder if it makes sense to use the `required` function here, e.g. 

```
{{ required "An AWS Access Token is required" .Values.faaslambda.access_token }}
```

A little preemptive validation so that users don't need to debug a crashloop or some other error via k8s logs.

Reply:
Sounds good to me..

@ewilde ?